_posts/2020-01-24-inception-pipeline-cdk.md

```diff
@@ -0,0 +1,395 @@
+---
+layout: post
+title: Cross-Account Inception Pipeline with AWS CDK
+date: 2020-01-20
+tags: codepipeline cloud-development-kit infrastructure-as-code aws cdk cross-account
+author: JK Gunnink
+image: img/inception-pipelines/seed_germination.png
+---
+
+The [inception pipeline][inception-pipeline-post] which Mechanical Rock's very own Pete Yandell
+pioneered back in early 2018 was created as a way to manage a pipeline through code, which even has
+the ability to update itself through CloudFormation steps.
+
+Using this for a few projects over the past 18 months or so has highlighted its value as projects
+develop, more steps are added, additional permissions are required or locked down and once you have
+a few resources, it quickly becomes a very large YAML file with over 1,000 lines of text. Anyone
+who's ever written large CloudFormation files will understand the pains of making sure resources are
+lined up and will be created and updated in the way you expect. Yes there are linting tools and
+other developer tools which can help make sure resources are referenced correctly etc, until you go
+to deploy. Enter: [Cloud Development Kit](https://aws.amazon.com/cdk/).
+
+## What is the CDK?
+
+The CDK is a relatively new tool from AWS which enables developers to build infrastructure at insane
+speed compared to more historic/traditional methods of writing out lines and lines of YAML or JSON.
+Using programming languages such as Typescript or Python, developers can make use of functions to
+scaffold code quickly, and under the hood the toolkit implements best-practices of cloud development
+which, for example is the principle of least privilege.
```

Comment:
Grammar seems a bit of, could rewrite to:
...implements the best-practices of cloud development which include, for example, the principle of least privilege.

Reply:
Being a pedant here, but you don't really need to mention 'cloud development' as principle of least privilege is not limited to cloud.

Reply:
Resolved here ed2d5fd6c20cc6ec1ec98e35d85dd77d7ded7323